File: converted/utilities/js-base64/elide-base64.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-base64

This demonstrates calling the TypeScript base64 implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One base64 implementation shared across TypeScript and Python
- Consistent base64 encoding behavior across services
- No Python library inconsistencies
- Perfect for data pipelines and processing
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# base64_module = require('./elide-base64.ts')

print("=== Python Calling TypeScript Base64 ===\n")

# Example 1: Basic Usage
# result = base64_module.default(input_data)
# print(f"Result: {result}")
# print()

# Example 2: Data Pipeline
# class DataProcessor:
#     def __init__(self):
#         self.base64 = base64_module
#
#     def process_batch(self, items):
#         """Process batch using base64"""
#         return [self.base64.default(item) for item in items]
#
# processor = DataProcessor()
# data = ['item1', 'item2', 'item3']
# results = processor.process_batch(data)
# for i, result in enumerate(results, 1):
#     print(f"  Item {i}: {result}")
# print()

print("Real-world use case:")
print("- Python data pipeline uses base64")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent base64 encoding across services")
print("- Encode binary data for APIs")
print()

print("Example: Cross-Service Architecture")
print("┌─────────────────────────────────────────┐")
print("│   Elide Base64 (TypeScript)           │")
print("│   elide-base64.ts                        │")
print("└─────────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same base64 encoding everywhere!")
print()

print("Problem Solved:")
print("Before: Python + JavaScript = different base64 encoding implementations")
print("After: One Elide implementation = 100% consistent behavior")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  base64 = require('./elide-base64.ts')")
print("  result = base64.default(data)")

File: converted/utilities/js-base64/elide-base64.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-base64
#
# This demonstrates calling the TypeScript base64 implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One base64 implementation shared across TypeScript and Ruby
# - Consistent base64 encoding behavior across microservices
# - No Ruby library discrepancies
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript Base64 ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# base64_module = Elide.require('./elide-base64.ts')

# Example 1: Basic Usage
# result = base64_module.default(input_data)
# puts "Result: #{result}"
# puts ""

# Example 2: Rails Integration
# class DataService
#   def initialize
#     @base64 = base64_module
#   end
#
#   def process(data)
#     @base64.default(data)
#   end
# end
#
# service = DataService.new
# result = service.process(input_data)
# puts "Processed: #{result}"
# puts ""

# Example 3: Sidekiq Worker
# class ProcessingJob
#   include Sidekiq::Worker
#
#   def perform(data)
#     result = base64_module.default(data)
#     puts "Job result: #{result}"
#   end
# end

puts "Real-world use case:"
puts "- Ruby worker uses base64"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent base64 encoding across services"
puts "- Encode binary data for APIs"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────────┐"
puts "│   Elide Base64 (TypeScript)           │"
puts "│   elide-base64.ts                        │"
puts "└─────────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same base64 encoding everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby + JavaScript = different implementations"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/base64.rb"
puts "  BASE64 = Elide.require('./elide-base64.ts')"
puts "  "
puts "  # app/services/data_service.rb"
puts "  class DataService"
puts "    def process(data)"
puts "      BASE64.default(data)"
puts "    end"
puts "  end"
